app/src/test/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityLocalTest.kt

```diff
@@ -247,6 +259,423 @@ class QuestionPlayerActivityLocalTest {
+
+      rotateToLandscape()
+
+      testCoroutineDispatchers.advanceTimeBy(TimeUnit.SECONDS.toMillis(30))
+      onView(withId(R.id.dot_hint)).check(
+        matches(isDisplayed())
+      )
+    }
+  }
+
+  @Test
+  fun testQuestionPlayer_submitOneWrongAnswer_hintAndSolNotVisibleVisible() {
+    launchForQuestionPlayer(SKILL_ID_LIST).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.question_recycler_view)).check(matches(isDisplayed()))
+
+      submitWrongAnswerToQuestionPlayerFractionInput()
+      onView(withId(R.id.hints_and_solution_fragment_container)).check(
+        matches(not(isDisplayed()))
+      )
+    }
+  }
+
+  @Test
+  fun testQuestionPlayer_submitOneWrongAnswer_wait60Seconds_hintAndSolVisibleVisible() {
+    launchForQuestionPlayer(SKILL_ID_LIST).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.question_recycler_view)).check(matches(isDisplayed()))
+
+      submitWrongAnswerToQuestionPlayerFractionInput()
+      onView(withId(R.id.hints_and_solution_fragment_container)).check(
+        matches(not(isDisplayed()))
+      )
+    }
+  }
+
+  @Test
+  fun testQuestionPlayer_submitTwoWrongAnswers_dotHintIconIsVisible() {
+    launchForQuestionPlayer(SKILL_ID_LIST).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.question_recycler_view)).check(matches(isDisplayed()))
+
+      submitTwoWrongAnswersToQuestionPlayer()
+      onView(withId(R.id.dot_hint)).check(
+        matches(isDisplayed())
+      )
+    }
+  }
+
+  @Test
+  fun testQuestionPlayer_hintConsumed_submitWrongAnswer_dotHintIconVisible() {
+    launchForQuestionPlayer(SKILL_ID_LIST).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.question_recycler_view)).check(matches(isDisplayed()))
+
+      makeFirstNewHintVisible()
+      openHintsAndSolutionDialog()
+      clickRevealNewHintAndSolution(hintAndSolutionIndex = 0, isSolution = false)
+      navigateBackToQuestionPlayer()
+      submitWrongAnswerToQuestionPlayerFractionInput()
+      onView(withId(R.id.dot_hint)).check(
+        matches(not(isDisplayed()))
+      )
+    }
+  }
+
+  @Test
+  fun testQuestionPlayer_hintConsumed_wait30Seconds_dotHintIconIsVisible() {
+    launchForQuestionPlayer(SKILL_ID_LIST).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.question_recycler_view)).check(matches(isDisplayed()))
+
+      makeFirstNewHintVisible()
+      openHintsAndSolutionDialog()
+      clickRevealNewHintAndSolution(hintAndSolutionIndex = 0, isSolution = false)
+      navigateBackToQuestionPlayer()
+      testCoroutineDispatchers.runCurrent()
+      testCoroutineDispatchers.advanceTimeBy(TimeUnit.SECONDS.toMillis(30))
+      onView(withId(R.id.dot_hint)).check(
+        matches(isDisplayed())
+      )
+    }
+  }
+
+  @Test
+  fun testQuestionPlayer_hintConsumed_submitWrongAnswer_wait10Seconds_dotHintIconIsVisible() {
```

Comment:
test name should be more clear, something like `firstHintConsumed` and `submitThirdWrongAnswer`

Reply:
adding both the action makes it go over 100 chars, changed the function name to `testQuestionPlayer_consumeFirstHint_submitWrongAnswer_wait10Seconds_dotHintIconIsVisible`